qiskit_experiments/curve_analysis/curve_analysis.py

```diff
@@ -804,70 +679,74 @@ def _run_analysis(
             data_processor.train(data=experiment_data.data())
 
         #
-        # 3. Extract curve entries from experiment data
+        # 2. Extract curve entries from experiment data
         #
         self._extract_curves(experiment_data=experiment_data, data_processor=data_processor)
 
         #
-        # 4. Run fitting
+        # 3. Run fitting
         #
         formatted_data = self._data(label="fit_ready")
+        fixed_params = {p: self.options.get(p) for p in self.__fixed_parameters__}
 
-        # Generate algorithmic initial guesses and boundaries
-        default_fit_opt = FitOptions(
-            parameters=self._fit_params(),
-            default_p0=self.options.p0,
-            default_bounds=self.options.bounds,
-            **self.options.curve_fitter_options,
-        )
+        fit_results = []
+        for fit_func in self.composite_funcs:
+            # Set parameter and data index to the composite fit function
+            if fixed_params:
+                fit_func.bind_parameters(**fixed_params)
+            fit_func.data_index = formatted_data.data_index
+
+            # Generate algorithmic initial guesses and boundaries
+            default_fit_opt = FitOptions(
+                group=fit_func.group,
+                parameters=fit_func.signature,
+                default_p0=self.options.p0,
+                default_bounds=self.options.bounds,
+                **self.options.curve_fitter_options,
+            )
 
-        fit_options = self._generate_fit_guesses(default_fit_opt)
-        if isinstance(fit_options, FitOptions):
-            fit_options = [fit_options]
+            fit_options = self._generate_fit_guesses(default_fit_opt)
+            if isinstance(fit_options, FitOptions):
+                fit_options = [fit_options]
 
-        # Run fit for each configuration
-        fit_results = []
-        for fit_opt in set(fit_options):
-            try:
-                fit_result = self.options.curve_fitter(
-                    funcs=[series_def.fit_func for series_def in self.__series__],
-                    series=formatted_data.data_index,
-                    xdata=formatted_data.x,
-                    ydata=formatted_data.y,
-                    sigma=formatted_data.y_err,
-                    **fit_opt.options,
+            # Run fit for each configuration
+            temp_results = []
+            for fit_opt in set(fit_options):
```

Comment:
Yes, that makes sense but parallelization is big pain in Qiskit due to multiple platform support. In addition, we already have composite analysis that spawns multiple threads, thus implementation might be tough. The standard Qiskit parallel framework doesn't fit since fit functions are not picklable @mtreinish ?